internal/config/from_env_test.go

```diff
@@ -107,38 +142,147 @@ func TestInheritEnv(t *testing.T) {
 			},
 		},
+		{
+			name: "GH_TOKEN shadows token from file",
+			baseConfig: heredoc.Doc(`
+			hosts:
+			  github.com:
+			    oauth_token: OTOKEN
+			`),
+			GH_TOKEN: "ENVTOKEN",
+			hostname: "github.com",
+			wants: wants{
+				hosts:     []string{"github.com"},
+				token:     "ENVTOKEN",
+				source:    "GH_TOKEN",
+				writeable: false,
+			},
+		},
+		{
+			name: "GITHUB_ENTERPRISE_TOKEN shadows token from file",
+			baseConfig: heredoc.Doc(`
+			hosts:
+			  example.org:
+			    oauth_token: OTOKEN
+			`),
+			GITHUB_ENTERPRISE_TOKEN: "ENVTOKEN",
+			hostname:                "example.org",
+			wants: wants{
+				hosts:     []string{"example.org"},
+				token:     "ENVTOKEN",
+				source:    "GITHUB_ENTERPRISE_TOKEN",
+				writeable: false,
+			},
+		},
+		{
+			name: "GH_ENTERPRISE_TOKEN shadows token from file",
+			baseConfig: heredoc.Doc(`
+			hosts:
+			  example.org:
+			    oauth_token: OTOKEN
+			`),
+			GH_ENTERPRISE_TOKEN: "ENVTOKEN",
+			hostname:            "example.org",
+			wants: wants{
+				hosts:     []string{"example.org"},
+				token:     "ENVTOKEN",
+				source:    "GH_ENTERPRISE_TOKEN",
+				writeable: false,
+			},
+		},
+		{
+			name:         "GH_TOKEN shadows token from GITHUB_TOKEN",
+			baseConfig:   ``,
+			GH_TOKEN:     "GHTOKEN",
+			GITHUB_TOKEN: "GITHUBTOKEN",
+			hostname:     "github.com",
+			wants: wants{
+				hosts:     []string{"github.com"},
+				token:     "GHTOKEN",
+				source:    "GH_TOKEN",
+				writeable: false,
+			},
+		},
+		{
+			name:                    "GH_ENTERPRISE_TOKEN shadows token from GITHUB_ENTERPRISE_TOKEN",
+			baseConfig:              ``,
+			GH_ENTERPRISE_TOKEN:     "GHTOKEN",
+			GITHUB_ENTERPRISE_TOKEN: "GITHUBTOKEN",
+			hostname:                "example.org",
+			wants: wants{
+				hosts:     []string(nil),
+				token:     "GHTOKEN",
+				source:    "GH_ENTERPRISE_TOKEN",
+				writeable: false,
+			},
+		},
 		{
 			name: "GITHUB_TOKEN adds host entry",
 			baseConfig: heredoc.Doc(`
 			hosts:
 			  example.org:
 			    oauth_token: OTOKEN
 			`),
-			GITHUB_TOKEN:            "ENVTOKEN",
-			GITHUB_ENTERPRISE_TOKEN: "",
-			hostname:                "github.com",
+			GITHUB_TOKEN: "ENVTOKEN",
+			hostname:     "github.com",
 			wants: wants{
 				hosts:     []string{"github.com", "example.org"},
 				token:     "ENVTOKEN",
 				source:    "GITHUB_TOKEN",
 				writeable: false,
 			},
 		},
+		{
+			name: "GH_TOKEN adds host entry",
+			baseConfig: heredoc.Doc(`
+			hosts:
+			  example.org:
+			    oauth_token: OTOKEN
+			`),
+			GH_TOKEN: "ENVTOKEN",
+			hostname: "github.com",
+			wants: wants{
+				hosts:     []string{"github.com", "example.org"},
+				token:     "ENVTOKEN",
+				source:    "GH_TOKEN",
+				writeable: false,
+			},
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			os.Setenv("GITHUB_TOKEN", tt.GITHUB_TOKEN)
-			os.Setenv("GITHUB_ENTERPRISE_TOKEN", tt.GITHUB_ENTERPRISE_TOKEN)
+			if tt.GITHUB_TOKEN != "" {
```

Comment:
Depending on where we end up with the new logic (see my previous comment), I think there would be value in allowing the tests setup to either set a variable to a blank value, or unset it completely. I'm not sure how we could express those conditions in table-driven test; perhaps:

```go
GH_TOKEN:     "(unset)"
GITHUB_TOKEN: "mytoken"
```

Reply:
Since we decided that unset and blank are equivalent I think we should be okay here.